fe/fe-core/src/test/java/org/apache/doris/rewrite/RewriteSelfCmpRuleTest.java

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.package org.apache.doris.rewrite;
+
+import org.apache.doris.common.FeConstants;
+import org.apache.doris.common.Pair;
+import org.apache.doris.utframe.DorisAssert;
+import org.apache.doris.utframe.UtFrameUtils;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+public class RewriteSelfCmpRuleTest {
```

Comment:
according @qidaye 's comment. we need to add test case that have expression in select list. and make sure all  `=`, `<`, `>`, `<=`, `>=` and `!=`  return `null` rather than `0`

Reply:
Good catch.

This convert is different for `select expr` and `where expr`

```sql
test=# select * from scientist
;
 id | firstname | lastname
----+-----------+----------
  1 | albert    | einstein
  2 | isaac     | newton
  3 | marie     | curie
    |           |
(4 rows)

test=# select id = id from scientist;
 ?column?
----------
 t
 t
 t

(4 rows)

test=# select id from scientist where id = id;
 id
----
  1
  2
  3
(3 rows)
```